observability/tracing/README.md

```diff
@@ -0,0 +1,103 @@
+# Distributed tracing
+
+With program flow distributed across many microservices, the ability to observe the different services in context can become a valuable tool for development and operations. 
+
+The simplest way to get an overview is to observe the Istio service mesh using Kiali. If there is a need to observe the causal relationship among individual calls to services, [envoy-based tracing](#envoy-based-context-extraction) is required.
+
+## Visualizing service mesh with Kiali
+
+Kiali can be reached at kiali.*cluster-name*.nais.io, eg. [kiali.dev-gcp.nais.io](https://kiali.dev-gcp.nais.io).
+
+![Kiali service mesh showing the relationship between sosialhjelp-modia, modia-api, and mock-alt-api](kiali-sample.gif)
+
+NAIS leverages the Istio service mesh and Kiali dashboard to give developers a visualization of the service mesh and its general health. Kiali will also give a quick graphical overview of failing HTTP calls.
+
+![Example of a service returning 400 errors](kiali-400-sample.gif)
+
+## Envoy-based context extraction
+
+Kiali offers a nice view of statistics. However, it is not possible to say that request A is a consequence of request B without adding additional context. Istio supports transparently reading special tracing headers from HTTP requests, and submitting trace data to jaeger.
+
+![Illustration of envoy-based tracing](envoy-tracing.png)
+
+This allows you to get up and running quickly with distributed tracing with relatively minimal changes needed to your code. All you need to do is copy the [appropriate headers](#propagating-trace-headers) from incoming requests to any outgoing it may cause, and istio-proxy and Jaeger does the rest for you.
+
+Loading a React page which issues hundreds of API calls generates a trace like this:
+
+![Example trace of a sosialhjelp-modia page load](example-trace.png)
+
+If envoy-based tracing covers your needs, this is the recommended approach to tracing. 
+But the approach does have some limitations; if you need to attach debug context or to create spans even inside a single node process, see [Direct submission](#direct-submission).
```

Comment:
Consider splitting up sentence for ease-of-mental-parsing?

Reply:
Done!